src/DetailsView/components/load-assessment-data-schema-provider.ts

```diff
@@ -0,0 +1,199 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { Assessment } from 'assessments/types/iassessment';
+
+export class LoadAssessmentDataSchemaProvider {
+    public getAssessmentSchema(assessments: readonly Readonly<Assessment>[]): any {
+        let schema = this.getAssessmentSchemaBase();
+
+        assessments.forEach(assessment => {
+            schema = this.setAssessmentBaseProperties(schema, assessment.key);
+
+            assessment.requirements.forEach(requirement => {
+                schema = this.setRequirementBaseProperties(schema, requirement.key, assessment.key);
+            });
+        });
+
+        schema = this.setDeprecatedRequirementProperties(schema);
+
+        return schema;
+    }
+
+    private setDeprecatedRequirementProperties(schema: any) {
+        const deprecatedRequirements = [
+            { assessmentKey: 'automated-checks', requirementKey: 'duplicate-id' },
+            { assessmentKey: 'automated-checks', requirementKey: 'scrollable-region-focusable' },
+        ];
+        deprecatedRequirements.forEach(requirement => {
+            if (
+                schema.properties.assessmentData.properties.assessments.properties[
+                    requirement.assessmentKey
+                ] === undefined
+            ) {
+                schema = this.setAssessmentBaseProperties(schema, requirement.assessmentKey);
+            }
+
+            schema = this.setRequirementBaseProperties(
+                schema,
+                requirement.requirementKey,
+                requirement.assessmentKey,
+            );
+        });
+        return schema;
+    }
+
+    private setAssessmentBaseProperties(schema: any, assessmentKey: string) {
+        const assessments = schema.properties.assessmentData.properties.assessments.properties;
```

Comment:
nit: this chain is so long and used so frequently, I almost think it might be worth writing a private "getAssessments(schema)" method that does this for you, to make accessing this more readable.